examples/seq2array/polygon/publisher/application.hpp

```diff
@@ -0,0 +1,141 @@
+/*
+* (c) Copyright, Real-Time Innovations, 2022.  All rights reserved.
+* RTI grants Licensee a license to use, modify, compile, and create derivative
+* works of the software solely for use with RTI Connext DDS. Licensee may
+* redistribute copies of the software provided that all such copies are subject
+* to this license. The software is provided "as is", with no warranty of any
+* type, including any warranty for fitness for any purpose. RTI is under no
+* obligation to maintain or support the software. RTI shall not be liable for
+* any incidental or consequential damages arising out of the use or inability
+* to use the software.
+*/
+
+#ifndef APPLICATION_HPP
+#define APPLICATION_HPP
+
+#include <iostream>
+#include <csignal>
+#include <dds/core/ddscore.hpp>
+
+namespace application {
+
+    // Catch control-C and tell application to shut down
+    bool shutdown_requested = false;
+
+    inline void stop_handler(int)
+    {
+        shutdown_requested = true;
+        std::cout << "preparing to shut down..." << std::endl;
+    }
+
+    inline void setup_signal_handlers()
+    {
+        signal(SIGINT, stop_handler);
+        signal(SIGTERM, stop_handler);
+    }
```

Comment:
this is generated by codegen, I pushed the file in order to avoid generating code with '-example <arch>' (that also made to set globally the ARCH, and generates a lot of unnecessary files).

Probably I can just create a publisher application without the 'application.hpp'.